pallets/disputes/src/lib.rs

```diff
@@ -0,0 +1,382 @@
+                    )
+                } else {
+                    Err(Error::<T>::DisputeDoesNotExist.into())
+                }
+            })?;
+
+            if votes.len() == total_jury {
+                if votes.iter().all(|v|*v.1) {
+                    Dispute::<T>::try_finalise_with_result(dispute_key, DisputeResult::Success)?;
+                }
+    
+                if votes.iter().all(|v|!*v.1) {
+                    Dispute::<T>::try_finalise_with_result(dispute_key, DisputeResult::Failure)?;
+                }
+            }
+            
+            Self::deposit_event(Event::<T>::DisputeVotedOn { who });
+            Ok(().into())
+        }
+
+        /// Force a dispute to fail. 
+        /// Must be called by T::ForceOrigin
+        #[pallet::call_index(1)]
+        #[pallet::weight(<T as Config>::WeightInfo::force_fail_dispute())]
+        pub fn force_fail_dispute(
+            origin: OriginFor<T>,
+            dispute_key: T::DisputeKey,
+        ) -> DispatchResult {
+            todo!();
+            Ok(().into())
+        }
+
+        /// Force a dispute to pass.
+        /// Must be called by T::ForceOrigin.
+        #[pallet::call_index(2)]
+        #[pallet::weight(<T as Config>::WeightInfo::force_succeed_dispute())]
+        pub fn force_succeed_dispute(
+            origin: OriginFor<T>,
+            dispute_key: T::DisputeKey,
+        ) -> DispatchResult {
+            todo!();
+            Ok(().into())
+        }
+
+        /// Extend a given dispute by T::VotingTimeLimit.
+        /// This can only be called once and must be called by a member of the jury.
+        #[pallet::call_index(3)]
+        #[pallet::weight(<T as Config>::WeightInfo::extend_dispute())]
+        pub fn extend_dispute(origin: OriginFor<T>, dispute_key: T::DisputeKey) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+            let mut dispute =
+                Disputes::<T>::get(dispute_key).ok_or(Error::<T>::DisputeDoesNotExist)?;
+            ensure!(!dispute.is_extended, Error::<T>::DisputeAlreadyExtended);
+            ensure!(
+                dispute.jury.iter().any(|e| e == &who),
+                Error::<T>::NotAJuryAccount
+            );
+            // Ensure that the dispute does not end on the old date.
+            DisputesFinaliseOn::<T>::mutate(dispute.expiration, |finalising| {
+                // TODO: see if this works lol
+                let _ = finalising
+                    .iter()
+                    .filter(|finalising_id| **finalising_id == dispute_key)
+                    .collect::<Vec<_>>();
+            });
+
+            // Insert the new date.
+            let new_expiry = dispute.expiration.saturating_add(T::VotingTimeLimit::get());
+            DisputesFinaliseOn::<T>::try_mutate(new_expiry, |finalising| {
+                finalising
+                    .try_push(dispute_key)
+                    .map_err(|_| Error::<T>::TooManyDisputesThisBlock)?;
```

Comment:
This also needs testing

Reply:
Thanks for the hint @f-gate, was able to test this from extend_dispute as well.